### PR TITLE
Update and reorganize README.md

### PR DESCRIPTION
**Pull request by odow**
x-ref https://github.com/jump-dev/JuMP.jl/pull/3624

I've made quite a few changes, so it's probably best viewed https://github.com/odow/Optim.jl/blob/patch-1/README.md

Main changes are:

 * Remove table of badges
 * Use explicit Markdown links
 * Remove mathematical description of rosebbrock
 * Remove suggestion to tag `@pfkod` explicitly
 * Fix JuMP example, which @blegat got wrong 😆 (missing objective sense)

### CODE REVIEW

**Comment by odow** [comment]
Will look like https://jump.dev/JuMP.jl/previews/PR3624/packages/Optim/

File: README.md [hunk 1]
```diff
@@ -1,73 +1,87 @@
...
+## Help and support
 
-| **Documentation**  | **Build Status** | **Reference to cite** |
-|:-:|:-:|:-:|
-| [![][docs-stable-img]][docs-stable-url]  | [![Build Status][build-linux-img]][build-linux-url] | [![JOSS][joss-img]][joss-url] |
-|  |[![Build Status][build-mac-img]][build-mac-url] |  | 
-|  |[![Build Status][build-windows-img]][build-windows-url] |  | 
-| |[![Codecov branch][cov-img]][cov-url]  | |
+For help and support, please post on the [Optimization (Mathematical)](https://discourse.julialang.org/c/domain/opt/13)
+section of the Julia discourse or the `#optimization` channel of the Julia [slack](https://julialang.org/slack/).
```
Comment (pkofod) [inline README.md:19]:
```suggestion
section of the Julia discourse or the `#math-optimization` channel of the Julia [slack](https://julialang.org/slack/).
```

**Comment by pkofod** [comment]
Looks good. I did suggest a change, because the slack channel name has math- first

**Comment by pkofod** [comment]
Thanks

**Pull request merged**